app.py

```diff
@@ -107,14 +207,132 @@ def do_logout(request):
 
 @view_config(route_name="question", renderer='templates/questionpage.jinja2')
 def question(request):
-    return {}
+    if request.authenticated_userid:
+        user = User.get_by_username(request.authenticated_userid)
+        if request.method == "POST":
+            answer = request.params.get("answer")
+            question = Question.get_question_by_id(
+                request.params.get("question_id")
+            )
+            if (answer not in [None, ''] and question.id not in
+               [sub.question_id for sub in Submission.get_all_for_user(user)]):
+
+                Submission.new(
+                    user=user,
+                    question=question,
+                    answer=answer
+                )
+        questions = Question.all()
+        if questions:
+            submissions = Submission.get_all_for_user(user)
+            l = []
+            for q in questions:
+                if q.id not in [s.question_id for s in submissions]:
+                    l.append(q)
```

Comment:
I'm feeling like there must be a way to get the list of all questions a user has not yet answered other than this.  The question to ask sql is "show me all questions where there is not a submission for that question from this user."  Doing this would provide a way to avoid one query and a full iteration through the list of questions.